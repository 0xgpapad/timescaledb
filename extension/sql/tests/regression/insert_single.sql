\set ON_ERROR_STOP 1

\set ECHO ALL
\ir include/insert_single.sql

\c single
\d+ "testNs".*
SELECT *
FROM "testNs"._hyper_1_0_distinct;
SELECT * FROM "testNs";
