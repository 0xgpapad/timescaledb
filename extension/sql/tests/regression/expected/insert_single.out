\ir include/insert_single.sql
\set ON_ERROR_STOP 1
\ir create_single_db.sql
SET client_min_messages = WARNING;
DROP DATABASE IF EXISTS single;
SET client_min_messages = NOTICE;
CREATE DATABASE single;
\c single
CREATE EXTENSION IF NOT EXISTS iobeamdb CASCADE;
psql:include/create_single_db.sql:7: NOTICE:  installing required extension "dblink"
psql:include/create_single_db.sql:7: NOTICE:  installing required extension "postgres_fdw"
psql:include/create_single_db.sql:7: NOTICE:  installing required extension "hstore"
SELECT setup_meta();
 setup_meta 
------------
 
(1 row)

SELECT setup_main();
 setup_main 
------------
 
(1 row)

\c single
\set ECHO ALL
SELECT add_cluster_user('postgres', NULL);
 add_cluster_user 
------------------
 
(1 row)

SELECT set_meta('single' :: NAME, 'fakehost');--fakehost makes sure there is no network communication
 set_meta 
----------
 
(1 row)

SELECT add_node('single' :: NAME, 'fakehost');
 add_node 
----------
 
(1 row)

CREATE TABLE PUBLIC."testNs" (
  "timeCustom" BIGINT NOT NULL,
  device_id TEXT NOT NULL,
  series_0 DOUBLE PRECISION NULL,
  series_1 DOUBLE PRECISION NULL,
  series_2 DOUBLE PRECISION NULL,
  series_bool BOOLEAN NULL
);
CREATE INDEX ON PUBLIC."testNs" (device_id, "timeCustom" DESC NULLS LAST) WHERE device_id IS NOT NULL;
CREATE INDEX ON PUBLIC."testNs" ("timeCustom" DESC NULLS LAST, series_0) WHERE series_0 IS NOT NULL;
CREATE INDEX ON PUBLIC."testNs" ("timeCustom" DESC NULLS LAST, series_1)  WHERE series_1 IS NOT NULL;
CREATE INDEX ON PUBLIC."testNs" ("timeCustom" DESC NULLS LAST, series_2) WHERE series_2 IS NOT NULL;
CREATE INDEX ON PUBLIC."testNs" ("timeCustom" DESC NULLS LAST, series_bool) WHERE series_bool IS NOT NULL;
SELECT * FROM create_hypertable('"public"."testNs"', 'timeCustom', 'device_id', hypertable_name=>'testNs', associated_schema_name=>'testNs' );
 create_hypertable 
-------------------
 
(1 row)

SELECT set_is_distinct_flag('"public"."testNs"', 'device_id', TRUE);
 set_is_distinct_flag 
----------------------
 
(1 row)

BEGIN;
\COPY "testNs" FROM 'data/ds1_dev1_1.tsv' NULL AS '';
COMMIT;
SELECT _iobeamdb_meta_api.close_chunk_end_immediate(c.id)
FROM get_open_partition_for_key('testNs', 'dev1') part
INNER JOIN _iobeamdb_catalog.chunk c ON (c.partition_id = part.id);
 close_chunk_end_immediate 
---------------------------
 
(1 row)

INSERT INTO "testNs"("timeCustom", device_id, series_0, series_1) VALUES
(1257987600000000000, 'dev1', 1.5, 1),
(1257987600000000000, 'dev1', 1.5, 2),
(1257894000000000000, 'dev20', 1.5, 1),
(1257894002000000000, 'dev1', 2.5, 3);
INSERT INTO "testNs"("timeCustom", device_id, series_0, series_1) VALUES
(1257894000000000000, 'dev20', 1.5, 2);
CREATE TABLE chunk_closing_test(
        time       BIGINT,
        metric     INTEGER,
        device_id  TEXT
    );
-- Test chunk closing/creation
SELECT * FROM create_hypertable('chunk_closing_test', 'time', 'device_id', chunk_size_bytes => 10000);
 create_hypertable 
-------------------
 
(1 row)

INSERT INTO chunk_closing_test VALUES(1, 1, 'dev1');
INSERT INTO chunk_closing_test VALUES(2, 2, 'dev2');
INSERT INTO chunk_closing_test VALUES(3, 3, 'dev3');
SELECT * FROM chunk_closing_test;
 time | metric | device_id 
------+--------+-----------
    1 |      1 | dev1
    2 |      2 | dev2
    3 |      3 | dev3
(3 rows)

SELECT * FROM _iobeamdb_catalog.chunk c
    LEFT JOIN _iobeamdb_catalog.chunk_replica_node crn ON (c.id = crn.chunk_id)
    LEFT JOIN _iobeamdb_catalog.partition_replica pr ON (crn.partition_replica_id = pr.id)
    WHERE hypertable_name = 'public.chunk_closing_test';
 id | partition_id | start_time | end_time | chunk_id | partition_replica_id | database_name |    schema_name     |     table_name      | id | partition_id |      hypertable_name      | replica_id |    schema_name     |       table_name       
----+--------------+------------+----------+----------+----------------------+---------------+--------------------+---------------------+----+--------------+---------------------------+------------+--------------------+------------------------
  3 |            2 |            |        1 |        3 |                    2 | single        | _iobeamdb_internal | _hyper_2_2_0_3_data |  2 |            2 | public.chunk_closing_test |          0 | _iobeamdb_internal | _hyper_2_2_0_partition
  4 |            2 |          2 |        2 |        4 |                    2 | single        | _iobeamdb_internal | _hyper_2_2_0_4_data |  2 |            2 | public.chunk_closing_test |          0 | _iobeamdb_internal | _hyper_2_2_0_partition
  5 |            2 |          3 |          |        5 |                    2 | single        | _iobeamdb_internal | _hyper_2_2_0_5_data |  2 |            2 | public.chunk_closing_test |          0 | _iobeamdb_internal | _hyper_2_2_0_partition
(3 rows)

\c single
\d+ "testNs".*
Index "testNs.1-testNs_device_id_timeCustom_idx"
   Column   |  Type  |  Definition  | Storage  
------------+--------+--------------+----------
 device_id  | text   | device_id    | extended
 timeCustom | bigint | "timeCustom" | plain
btree, for table "testNs._hyper_1_1_0_1_data", predicate (device_id IS NOT NULL)

Index "testNs.10-testNs_timeCustom_series_bool_idx"
   Column    |  Type   |  Definition  | Storage 
-------------+---------+--------------+---------
 timeCustom  | bigint  | "timeCustom" | plain
 series_bool | boolean | series_bool  | plain
btree, for table "testNs._hyper_1_1_0_2_data", predicate (series_bool IS NOT NULL)

    Index "testNs.2-testNs_timeCustom_series_0_idx"
   Column   |       Type       |  Definition  | Storage 
------------+------------------+--------------+---------
 timeCustom | bigint           | "timeCustom" | plain
 series_0   | double precision | series_0     | plain
btree, for table "testNs._hyper_1_1_0_1_data", predicate (series_0 IS NOT NULL)

    Index "testNs.3-testNs_timeCustom_series_1_idx"
   Column   |       Type       |  Definition  | Storage 
------------+------------------+--------------+---------
 timeCustom | bigint           | "timeCustom" | plain
 series_1   | double precision | series_1     | plain
btree, for table "testNs._hyper_1_1_0_1_data", predicate (series_1 IS NOT NULL)

    Index "testNs.4-testNs_timeCustom_series_2_idx"
   Column   |       Type       |  Definition  | Storage 
------------+------------------+--------------+---------
 timeCustom | bigint           | "timeCustom" | plain
 series_2   | double precision | series_2     | plain
btree, for table "testNs._hyper_1_1_0_1_data", predicate (series_2 IS NOT NULL)

Index "testNs.5-testNs_timeCustom_series_bool_idx"
   Column    |  Type   |  Definition  | Storage 
-------------+---------+--------------+---------
 timeCustom  | bigint  | "timeCustom" | plain
 series_bool | boolean | series_bool  | plain
btree, for table "testNs._hyper_1_1_0_1_data", predicate (series_bool IS NOT NULL)

Index "testNs.6-testNs_device_id_timeCustom_idx"
   Column   |  Type  |  Definition  | Storage  
------------+--------+--------------+----------
 device_id  | text   | device_id    | extended
 timeCustom | bigint | "timeCustom" | plain
btree, for table "testNs._hyper_1_1_0_2_data", predicate (device_id IS NOT NULL)

    Index "testNs.7-testNs_timeCustom_series_0_idx"
   Column   |       Type       |  Definition  | Storage 
------------+------------------+--------------+---------
 timeCustom | bigint           | "timeCustom" | plain
 series_0   | double precision | series_0     | plain
btree, for table "testNs._hyper_1_1_0_2_data", predicate (series_0 IS NOT NULL)

    Index "testNs.8-testNs_timeCustom_series_1_idx"
   Column   |       Type       |  Definition  | Storage 
------------+------------------+--------------+---------
 timeCustom | bigint           | "timeCustom" | plain
 series_1   | double precision | series_1     | plain
btree, for table "testNs._hyper_1_1_0_2_data", predicate (series_1 IS NOT NULL)

    Index "testNs.9-testNs_timeCustom_series_2_idx"
   Column   |       Type       |  Definition  | Storage 
------------+------------------+--------------+---------
 timeCustom | bigint           | "timeCustom" | plain
 series_2   | double precision | series_2     | plain
btree, for table "testNs._hyper_1_1_0_2_data", predicate (series_2 IS NOT NULL)

               Table "testNs._hyper_1_0_1_distinct_data"
   Column    | Type | Modifiers | Storage  | Stats target | Description 
-------------+------+-----------+----------+--------------+-------------
 column_name | text | not null  | extended |              | 
 value       | text | not null  | extended |              | 
Indexes:
    "_hyper_1_0_1_distinct_data_pkey" PRIMARY KEY, btree (column_name, value)
Inherits: "testNs"._hyper_1_0_distinct

Index "testNs._hyper_1_0_1_distinct_data_pkey"
   Column    | Type | Definition  | Storage  
-------------+------+-------------+----------
 column_name | text | column_name | extended
 value       | text | value       | extended
primary key, btree, for table "testNs._hyper_1_0_1_distinct_data"

                   Table "testNs._hyper_1_0_distinct"
   Column    | Type | Modifiers | Storage  | Stats target | Description 
-------------+------+-----------+----------+--------------+-------------
 column_name | text | not null  | extended |              | 
 value       | text | not null  | extended |              | 
Inherits: "testNs"._hyper_1_distinct
Child tables: "testNs"._hyper_1_0_1_distinct_data

                         Table "testNs._hyper_1_0_replica"
   Column    |       Type       | Modifiers | Storage  | Stats target | Description 
-------------+------------------+-----------+----------+--------------+-------------
 timeCustom  | bigint           | not null  | plain    |              | 
 device_id   | text             | not null  | extended |              | 
 series_0    | double precision |           | plain    |              | 
 series_1    | double precision |           | plain    |              | 
 series_2    | double precision |           | plain    |              | 
 series_bool | boolean          |           | plain    |              | 
Inherits: "testNs"._hyper_1_root
Child tables: "testNs"._hyper_1_1_0_partition

                         Table "testNs._hyper_1_1_0_1_data"
   Column    |       Type       | Modifiers | Storage  | Stats target | Description 
-------------+------------------+-----------+----------+--------------+-------------
 timeCustom  | bigint           | not null  | plain    |              | 
 device_id   | text             | not null  | extended |              | 
 series_0    | double precision |           | plain    |              | 
 series_1    | double precision |           | plain    |              | 
 series_2    | double precision |           | plain    |              | 
 series_bool | boolean          |           | plain    |              | 
Indexes:
    "1-testNs_device_id_timeCustom_idx" btree (device_id, "timeCustom" DESC NULLS LAST) WHERE device_id IS NOT NULL
    "2-testNs_timeCustom_series_0_idx" btree ("timeCustom" DESC NULLS LAST, series_0) WHERE series_0 IS NOT NULL
    "3-testNs_timeCustom_series_1_idx" btree ("timeCustom" DESC NULLS LAST, series_1) WHERE series_1 IS NOT NULL
    "4-testNs_timeCustom_series_2_idx" btree ("timeCustom" DESC NULLS LAST, series_2) WHERE series_2 IS NOT NULL
    "5-testNs_timeCustom_series_bool_idx" btree ("timeCustom" DESC NULLS LAST, series_bool) WHERE series_bool IS NOT NULL
Check constraints:
    "partition" CHECK (get_partition_for_key(device_id, 32768) >= '0'::smallint AND get_partition_for_key(device_id, 32768) <= '32767'::smallint)
    "time_range" CHECK ("timeCustom" <= '1257897600000000000'::bigint)
Inherits: "testNs"._hyper_1_1_0_partition

                         Table "testNs._hyper_1_1_0_2_data"
   Column    |       Type       | Modifiers | Storage  | Stats target | Description 
-------------+------------------+-----------+----------+--------------+-------------
 timeCustom  | bigint           | not null  | plain    |              | 
 device_id   | text             | not null  | extended |              | 
 series_0    | double precision |           | plain    |              | 
 series_1    | double precision |           | plain    |              | 
 series_2    | double precision |           | plain    |              | 
 series_bool | boolean          |           | plain    |              | 
Indexes:
    "10-testNs_timeCustom_series_bool_idx" btree ("timeCustom" DESC NULLS LAST, series_bool) WHERE series_bool IS NOT NULL
    "6-testNs_device_id_timeCustom_idx" btree (device_id, "timeCustom" DESC NULLS LAST) WHERE device_id IS NOT NULL
    "7-testNs_timeCustom_series_0_idx" btree ("timeCustom" DESC NULLS LAST, series_0) WHERE series_0 IS NOT NULL
    "8-testNs_timeCustom_series_1_idx" btree ("timeCustom" DESC NULLS LAST, series_1) WHERE series_1 IS NOT NULL
    "9-testNs_timeCustom_series_2_idx" btree ("timeCustom" DESC NULLS LAST, series_2) WHERE series_2 IS NOT NULL
Check constraints:
    "partition" CHECK (get_partition_for_key(device_id, 32768) >= '0'::smallint AND get_partition_for_key(device_id, 32768) <= '32767'::smallint)
    "time_range" CHECK ("timeCustom" >= '1257897600000000001'::bigint)
Inherits: "testNs"._hyper_1_1_0_partition

                       Table "testNs._hyper_1_1_0_partition"
   Column    |       Type       | Modifiers | Storage  | Stats target | Description 
-------------+------------------+-----------+----------+--------------+-------------
 timeCustom  | bigint           | not null  | plain    |              | 
 device_id   | text             | not null  | extended |              | 
 series_0    | double precision |           | plain    |              | 
 series_1    | double precision |           | plain    |              | 
 series_2    | double precision |           | plain    |              | 
 series_bool | boolean          |           | plain    |              | 
Check constraints:
    "partition" CHECK (get_partition_for_key(device_id, 32768) >= '0'::smallint AND get_partition_for_key(device_id, 32768) <= '32767'::smallint)
Inherits: "testNs"._hyper_1_0_replica
Child tables: "testNs"._hyper_1_1_0_1_data,
              "testNs"._hyper_1_1_0_2_data

                    Table "testNs._hyper_1_distinct"
   Column    | Type | Modifiers | Storage  | Stats target | Description 
-------------+------+-----------+----------+--------------+-------------
 column_name | text | not null  | extended |              | 
 value       | text | not null  | extended |              | 
Indexes:
    "_hyper_1_distinct_pkey" PRIMARY KEY, btree (column_name, value)
Child tables: "testNs"._hyper_1_0_distinct

    Index "testNs._hyper_1_distinct_pkey"
   Column    | Type | Definition  | Storage  
-------------+------+-------------+----------
 column_name | text | column_name | extended
 value       | text | value       | extended
primary key, btree, for table "testNs._hyper_1_distinct"

                            Table "testNs._hyper_1_root"
   Column    |       Type       | Modifiers | Storage  | Stats target | Description 
-------------+------------------+-----------+----------+--------------+-------------
 timeCustom  | bigint           | not null  | plain    |              | 
 device_id   | text             | not null  | extended |              | 
 series_0    | double precision |           | plain    |              | 
 series_1    | double precision |           | plain    |              | 
 series_2    | double precision |           | plain    |              | 
 series_bool | boolean          |           | plain    |              | 
Child tables: "testNs"._hyper_1_0_replica

SELECT *
FROM "testNs"._hyper_1_0_distinct;
 column_name | value 
-------------+-------
 device_id   | dev1
 device_id   | dev20
(2 rows)

SELECT * FROM "testNs";
     timeCustom      | device_id | series_0 | series_1 | series_2 | series_bool 
---------------------+-----------+----------+----------+----------+-------------
 1257894000000000000 | dev1      |      1.5 |        1 |        2 | t
 1257894000000000000 | dev1      |      1.5 |        2 |          | 
 1257894000000001000 | dev1      |      2.5 |        3 |          | 
 1257894001000000000 | dev1      |      3.5 |        4 |          | 
 1257897600000000000 | dev1      |      4.5 |        5 |          | f
 1257894000000000000 | dev20     |      1.5 |        1 |          | 
 1257894002000000000 | dev1      |      2.5 |        3 |          | 
 1257894000000000000 | dev20     |      1.5 |        2 |          | 
 1257987600000000000 | dev1      |      1.5 |        1 |          | 
 1257987600000000000 | dev1      |      1.5 |        2 |          | 
(10 rows)

